shell/renderer/electron_renderer_client.cc

```diff
@@ -109,20 +109,24 @@ void ElectronRendererClient::DidCreateScriptContext(
                        (is_not_opened || reuse_renderer_processes_enabled);
   bool is_devtools = IsDevToolsExtension(render_frame);
   bool allow_node_in_subframes = prefs.node_integration_in_sub_frames;
+  // If we're loading Node.js into a subframe and we've already initialized
+  // the integration, that means we're here because a new subframe was created,
+  // not because we've reloaded the page. In that case, don't re-prepare the
+  // message loop.
+  bool node_in_subframe = !is_main_frame && allow_node_in_subframes;
   bool should_load_node =
       (is_main_frame || is_devtools || allow_node_in_subframes) &&
       !IsWebViewFrame(renderer_context, render_frame);
-  if (!should_load_node) {
+  if (!should_load_node)
     return;
-  }
 
   injected_frames_.insert(render_frame);
 
   if (!node_integration_initialized_) {
     node_integration_initialized_ = true;
     node_bindings_->Initialize();
     node_bindings_->PrepareMessageLoop();
-  } else if (reuse_renderer_processes_enabled) {
+  } else if (reuse_renderer_processes_enabled && !node_in_subframe) {
```

Comment:
An edge case situation where one creates a window with only `nodeIntegrationInSubframes: true`, render process reuse enabled and the main frame doesn't have nodeIntegration. Now when they reload the page, the subframe node integration message loop will not work correctly https://github.com/electron/electron/pull/25869

`else if (reuse_renderer_processes_enabled` should not be based on frames but instead whether the page was loaded in the same process. Can be checked with browser navigation state or the message loop handles or renderer client state.